syntax = "proto3";

message TaskRequest {
  string codePath=1;    //  pantheon path where the input code is stored
  string dataPath=2;    //  pantheon path where the input files are stored
  string outputPath=3;  //  pantheon path where the output will be stored
  string targetPath=4;      //  path to file from code root that is to be executed
  double timeout=5;     //  timeout for the execution 
  repeated configPair actionPairs =6; //key value pairs containing action and related parameters
}
message configPair{ //key value pairs containing action and related parameters
  string key=1;
  string value=2;
}
message TaskResponse {
  string status=1;    
  int32 numberOfFiles=2;
  double timeTaken=5;
}
 